translib/transformer/test/transformer_test.go

```diff
@@ -0,0 +1,299 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//////////////////////////////////////////////////////////////////////////
+
+package transformer_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/Azure/sonic-mgmt-common/translib/db"
+	"github.com/Azure/sonic-mgmt-common/translib/ocbinds"
+	"github.com/go-redis/redis/v7"
+	"github.com/openconfig/ygot/ytypes"
+
+	"testing"
+)
+
+var dbConfig struct {
+	Instances map[string]map[string]interface{} `json:"INSTANCES"`
+	Databases map[string]map[string]interface{} `json:"DATABASES"`
+}
+var rclient *redis.Client
+var filehandle *os.File
+var ygSchema *ytypes.Schema
+var rclientDBNum map[db.DBNum]*redis.Client
+
+func getDBOptions(dbNo db.DBNum, isWriteDisabled bool) db.Options {
+        var opt db.Options
+
+        switch dbNo {
+        case db.ApplDB, db.CountersDB, db.AsicDB, db.FlexCounterDB:
+                opt = getDBOptionsWithSeparator(dbNo, "", ":", ":", isWriteDisabled)
+                break
+        case db.ConfigDB, db.StateDB, db.ErrorDB:
+                opt = getDBOptionsWithSeparator(dbNo, "", "|", "|", isWriteDisabled)
+                break
+        }
+
+        return opt
+}
+
+func getDBOptionsWithSeparator(dbNo db.DBNum, initIndicator string, tableSeparator string, keySeparator string, isWriteDisabled bool) db.Options {
+        return (db.Options{
+                DBNo:               dbNo,
+                InitIndicator:      initIndicator,
+                TableNameSeparator: tableSeparator,
+                KeySeparator:       keySeparator,
+                IsWriteDisabled:    isWriteDisabled,
+        })
+}
+
+func getAllDbs(isGetCase bool) ([db.MaxDB]*db.DB, error) {
+	var dbs [db.MaxDB]*db.DB
+	var err error
+	var isWriteDisabled bool
+
+	if isGetCase {
+		isWriteDisabled = true
+	} else {
+		isWriteDisabled = false
+	}
+
+	//Create Counter DB connection
+	dbs[db.CountersDB], err = db.NewDB(getDBOptions(db.CountersDB, isWriteDisabled))
+
+	if err != nil {
+		closeAllDbs(dbs[:])
+		return dbs, err
+	}
+        
+        //Create Config DB connection
+	dbs[db.ConfigDB], err = db.NewDB(getDBOptions(db.ConfigDB, isWriteDisabled))
+
+	if err != nil {
+		closeAllDbs(dbs[:])
+		return dbs, err
+	}
+
+	return dbs, err
+}
+
+// Closes the dbs, and nils out the arr.
+func closeAllDbs(dbs []*db.DB) {
+	for dbsi, d := range dbs {
+		if d != nil {
+			d.DeleteDB()
+			dbs[dbsi] = nil
+		}
+	}
+}
+
+func TestMain(t *testing.M) {
+	fmt.Println("----- Setting up transformer tests -----")
+	if err := setup(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error setting up transformer testing state: %v.\n", err)
+		os.Exit(1)
+	}
+	t.Run()
+	teardown()
+	os.Exit(0)
```

Comment:
With `Exit(0)`, shell will treat all tests as successful even when some of them fail.
Pleas use the return value of `t.Run()` as the exit code.

Reply:
Addressed as suggested

Reply:
I still see blind `os.Exit(0)`. Better to rearrange as:

```go
defer teardown()
os.Exit(t.Run())
```